### ISSUE TITLE
New cache pop up is crashing back to map

### ISSUE THREAD

**Issue by wolverine007** [cgeo/cgeo#15247]
### Describe your problem!

The new cache pop up box is crashing when being opened via swipe up.
Cgeo is shortly displaying the details and then going back to map. 

### How to reproduce?

open any cache by selecting it on the map

### Actual result after these steps?

Cache details are displayed shortly, then app is going back to map

### Expected result after these steps?

Show cache details like being done by clicking the according button. 

### Reproducible

Yes

### c:geo Version

2024.01.24-RC

### System information

```text
## System information

c:geo version: 2024.01.24-RC

Device:
-------
- Device type: 23049PCD8G (marble_eea, POCO)
- Available processors: 8
- Android version: 13
- Android build: TKQ1.221114.001 test-keys
- Screen resolution: 1080x2176px (392x791dp)
- Pixel density: 2.75
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 24.02.13 (190400-599553831)
- Memory:  Available:1,6 GB, Total:7,0 GB, Threshold: 216,0 MB, low:false

Sensor and location:
-------
- Low power mode: active
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: orientation

Program settings:
-------
- Settings: v9, Count:281
- Set language: de
- System date format: dd.MM.yy
- Time zone: GMT+01:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: never
- Last auto backup: 24. Jan., 09:32
- Routing mode: Bike
- Live map mode: true
- Use unified map: false
- OSM multi-threading: true / threads: 4
- Map: Deu+nord (FZK) (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:cgeo/maps/Deu+nord (FZK).map
  - Atts: (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti
  - Theme: fzk-outdoor-soft-v5.zip:fzk-outdoor-soft-v5/fzk-outdoor-soft-v5.xml

Filters:
-------
- Hide waypoints: visited
- LIVE: Status: 3 Elemente ({"advanced":false,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{},"t":{}}},{"type":"status","config":{"values":["owned_no","found_no","exclude_archived"]}}]}})
- OFFLINE: Keine ({"advanced":false,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{},"t":{}}},{"type":"status","config":{}}]}})


Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   - Geocaching.com Adventure Lab
- Geocaching.com date format: dd.MM.yyyy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2023-10-25 07-39)
- Last successful login on geocaching.com: 2024-01-26 23-10
- Routing: internal / BRouter installed: false

Permissions
-------
- ACCESS_COARSE_LOCATION: granted
- ACCESS_FINE_LOCATION: granted
- READ_CONTACTS: DENIED
- WRITE_EXTERNAL_STORAGE: DENIED
- POST_NOTIFICATIONS: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (168,4 GB free) v3 internal isDir(9 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (168,4 GB free) v3 internal isDir(9 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (168,4 GB free) v3 external non-removable isDir(1002 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (168,4 GB free) v3 internal isDir(9 entries)
- Map render theme path: fzk-outdoor-soft-v5.zip:fzk-outdoor-soft-v5/fzk-outdoor-soft-v5.xml
- Public Folders: #12
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=19, dirs:>=9, totalFileSize:>=5,0 GB, free space: 168,4 GB, files on device: 12812797)
  - OFFLINE_MAPS: /cgeo/maps (Default)[/cgeo/maps[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps, Av:true, files:17, dirs:2, totalFileSize:5,0 GB, free space: 168,4 GB, files on device: 12812797)
  - OFFLINE_MAP_THEMES: /cgeo/maps/_themes (Default)[/cgeo/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps%2F_themes, Av:true, files:8, dirs:0, totalFileSize:1,4 MB, free space: 168,4 GB, files on device: 12812797)
  - OFFLINE_MAP_SHADING: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=19, dirs:>=9, totalFileSize:>=5,0 GB, free space: 168,4 GB, files on device: 12812797)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 168,4 GB, files on device: 12812797)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:2, dirs:0, totalFileSize:807,6 KB, free space: 168,4 GB, files on device: 12812797)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=8, dirs:>=6, totalFileSize:>=1,0 GB, free space: 168,4 GB, files on device: 12812797)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 168,4 GB, files on device: 12812797)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 168,4 GB, files on device: 12812797)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:10, dirs:1, totalFileSize:263,1 MB, free space: 168,4 GB, files on device: 12812797)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:2, dirs:0, totalFileSize:263,0 MB, free space: 168,4 GB, files on device: 12812797)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 168,4 GB, files on device: -1)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (19. Nov. 2023, 16:41):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v103, Size:260,3 MB) on system internal storage

--- End of system information ---
```

### Additional Information

Error message being stated after the second crash:

java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1844)
	at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1884)
	at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:329)
	at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:294)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity.sheetRemoveFragment(AbstractNavigationBarMapActivity.java:155)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1.lambda$onStateChanged$0(AbstractNavigationBarMapActivity.java:109)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1.$r8$lambda$RpaYKj513Cn7DKz0HIXkkxb6ZTg(AbstractNavigationBarMapActivity.java:0)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:211)
	at android.os.Looper.loop(Looper.java:300)
	at android.app.ActivityThread.main(ActivityThread.java:8395)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)

**Comment by wolverine007** [comment]
https://github.com/cgeo/cgeo/assets/4508228/326f27d6-d37d-4a81-9c94-c4f9e94fcc35

1) Swipe Up
2) More Details (Mehr Details)

**Comment by moviti** [comment]
In my case, when I swipe up , apk will crash and force close.
android 14, pixel 8

**Comment by Lineflyer** [comment]
Interesting enough I cannot reproduce this and - so far - we got no similar complaints on support the our current beta version.
As this is a major issue...does anyone have a clue (from the stack trace above)?

**Comment by fm-sys** [comment]
```
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1844)
at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1884)
at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:329)
at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:294)
at cgeo.geocaching.activity.AbstractNavigationBarMapActivity.sheetRemoveFragment(AbstractNavigationBarMapActivity.java:155)
```

To me this looks like as if we are trying to remove a fragment which belongs to an activity which is already destroyed...

**Comment by moving-bits** [comment]
I've tried to reproduce this by setting "don't keep activities" in the device's developer settings, but was not able to reproduce this behavior.

**Comment by eddiemuc** [comment]
I can consistently reproduce on my phydsical device (OnePlus 7T, Android 12), but not on an emulator.

Googling this gives a lot of results e.g. https://stackoverflow.com/questions/7575921/illegalstateexception-can-not-perform-this-action-after-onsaveinstancestate-wit

I created PR #15294 which checks activity state before executing fragment removals. can't tell whether this solves the problem but it might be worth a try.

**Comment by fm-sys** [comment]
> I can consistently reproduce on my phydsical device (OnePlus 7T, Android 12), but not on an emulator.

You could install the debug build in parallel on your physical device and test it there. This is possible since a few weeks.

Alternatively we just merge and hope for the best. Looks quite promising...

**Comment by ztNFny** [comment]
> I created PR #15294 which checks activity state before executing fragment removals. can't tell whether this solves the problem but it might be worth a try.

I actually didn't have this crash last week but this week - with the PR included - it crashes reproducably.

java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1844)
	at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1884)
	at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:329)
	at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:294)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity.sheetRemoveFragment(AbstractNavigationBarMapActivity.java:172)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1.lambda$onStateChanged$0(AbstractNavigationBarMapActivity.java:122)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1.$r8$lambda$RpaYKj513Cn7DKz0HIXkkxb6ZTg(AbstractNavigationBarMapActivity.java:0)
	at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:233)
	at android.os.Looper.loop(Looper.java:344)
	at android.app.ActivityThread.main(ActivityThread.java:8212)

**Comment by eddiemuc** [comment]
> > I created PR #15294 which checks activity state before executing fragment removals. can't tell whether this solves the problem but it might be worth a try.
> 
> I actually didn't have this crash last week but this week - with the PR included - it crashes reproducably.
> 
> java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1844) at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1884) at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:329) at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:294) at cgeo.geocaching.activity.AbstractNavigationBarMapActivity.sheetRemoveFragment(AbstractNavigationBarMapActivity.java:172) at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1.lambda$onStateChanged$0(AbstractNavigationBarMapActivity.java:122) at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1.$r8$lambda$RpaYKj513Cn7DKz0HIXkkxb6ZTg(AbstractNavigationBarMapActivity.java:0) at cgeo.geocaching.activity.AbstractNavigationBarMapActivity$1$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0) at android.os.Handler.handleCallback(Handler.java:938) at android.os.Handler.dispatchMessage(Handler.java:99) at android.os.Looper.loopOnce(Looper.java:233) at android.os.Looper.loop(Looper.java:344) at android.app.ActivityThread.main(ActivityThread.java:8212)

It may be that the PR doesn't solve the problem unfortunately. Alas I am not an expert for fragments, maybe someone else can resolve this better.

What I don't understand however: the stack trace above claims that there is a call to `commit` in like `172 `in class file `AbstractNavigationBarMapActivity.java` as well as a call to method `sheetRemoveFragment `in line `122 `of same file. I am unable to match this to current states in master nor release. Which exact revisions were used to produce the stack trace above?

Line 172 in master: https://github.com/cgeo/cgeo/blob/a529a682ea3ac53f91bfbf82a2d120deed3a1db5/main/src/main/java/cgeo/geocaching/activity/AbstractNavigationBarMapActivity.java#L172 

Line 122 in master: https://github.com/cgeo/cgeo/blob/a529a682ea3ac53f91bfbf82a2d120deed3a1db5/main/src/main/java/cgeo/geocaching/activity/AbstractNavigationBarMapActivity.java#L122

Lines in release are the same

**Comment by ztNFny** [comment]
> What I don't understand however: the stack trace above claims that there is a call to `commit` in like `172 `in class file `AbstractNavigationBarMapActivity.java` as well as a call to method `sheetRemoveFragment `in line `122 `of same file. I am unable to match this to current states in master nor release. Which exact revisions were used to produce the stack trace above?

I have some modifications in my AbstractNavigationBarMapAcvtivity - but the crash is reproducable without those in master, I accidentally picked the wrong stacktrace when posting it.

**Comment by ztNFny** [comment]
Had some more time to play around a bit. Findings:

Even checking right before the commit doesn't solve this:
```
            if (isFinishing() || isDestroyed()) return false;
            fm.beginTransaction().remove(f1).commit();
```

Then I found the statement: "In general, the best way to avoid the exception in these cases is to simply avoid committing transactions in asynchronous callback methods all together."

the call is explicitly done asynchronous using ActivityMixin.postDelayed(), so probably the "good" solution would be rewriting this completly

Not wanting to do that that I tried commitAllowingStateLoss() instead of commit() - that resolves the crash. But ...
- it's use is discouraged (according to stackoverflow and co)
- ~500ms after opening CacheDetailsActivity the screen flickers (the removal action I guess). Acceptable but not nice.
